src/panels/config/integrations/integration-panels/bluetooth/dialog-bluetooth-device-info.ts

```diff
@@ -0,0 +1,125 @@
+import type { TemplateResult } from "lit";
+import { html, LitElement, nothing } from "lit";
+import { customElement, property, state } from "lit/decorators";
+import { fireEvent } from "../../../../../common/dom/fire_event";
+import type { HassDialog } from "../../../../../dialogs/make-dialog-manager";
+import { createCloseHeading } from "../../../../../components/ha-dialog";
+import type { HomeAssistant } from "../../../../../types";
+import type { BluetoothDeviceInfoDialogParams } from "./show-dialog-bluetooth-device-info";
+import "../../../../../components/ha-button";
+import { showToast } from "../../../../../util/toast";
+import { copyToClipboard } from "../../../../../common/util/copy-clipboard";
+
+@customElement("dialog-bluetooth-device-info")
+class DialogBluetoothDeviceInfo extends LitElement implements HassDialog {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @state() private _params?: BluetoothDeviceInfoDialogParams;
+
+  public async showDialog(
+    params: BluetoothDeviceInfoDialogParams
+  ): Promise<void> {
+    this._params = params;
+  }
+
+  public closeDialog(): void {
+    this._params = undefined;
+    fireEvent(this, "dialog-closed", { dialog: this.localName });
+  }
+
+  public showDataAsHex(bytestring: string): string {
+    return Array.from(new TextEncoder().encode(bytestring))
+      .map((byte) => byte.toString(16).toUpperCase().padStart(2, "0"))
+      .join(" ");
+  }
+
+  private async _copyToClipboard(): Promise<void> {
+    if (!this._params) {
```

Comment:
Since we don't render anything when we have no params, you can use `params!` below to tell TS that the variable is there.